src/ir_Corona.cpp

```diff
@@ -0,0 +1,578 @@
+            kCoronaAcFanOffset, kCoronaAcFanSize, speed);
+}
+
+/// Change the powersave setting.
+/// @param[in] on true, the setting is on. false, the setting is off.
+void IRCoronaAc::setEcono(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionData0Pos], kCoronaAcPowerSaveOffset, on);
+}
+
+/// Get the value of the current powersave setting.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getEcono(void) {
+  return GETBIT8(remote_state[kCoronaAcSectionData0Pos],
+                 kCoronaAcPowerSaveOffset);
+}
+
+/// Convert a standard A/C Fan speed into its native fan speed.
+/// @param[in] speed The desired stdAc::fanspeed_t fan speed
+/// @return The given fan speed in native format
+uint8_t IRCoronaAc::convertFan(const stdAc::fanspeed_t speed) {
+  switch (speed) {
+    case stdAc::fanspeed_t::kMin:
+    case stdAc::fanspeed_t::kLow:    return kCoronaAcFanLow;
+    case stdAc::fanspeed_t::kMedium: return kCoronaAcFanMedium;
+    case stdAc::fanspeed_t::kHigh:
+    case stdAc::fanspeed_t::kMax:    return kCoronaAcFanHigh;
+    default:                         return kCoronaAcFanAuto;
+  }
+}
+
+/// Convert a native fan speed to it's common equivalent.
+/// @param[in] speed The desired native fan speed
+/// @return The given fan speed in stdAc::fanspeed_t format
+stdAc::fanspeed_t IRCoronaAc::toCommonFanSpeed(const uint8_t speed) {
+  switch (speed) {
+    case kCoronaAcFanHigh:    return stdAc::fanspeed_t::kHigh;
+    case kCoronaAcFanMedium:  return stdAc::fanspeed_t::kMedium;
+    case kCoronaAcFanLow:     return stdAc::fanspeed_t::kLow;
+    default:                  return stdAc::fanspeed_t::kAuto;
+  }
+}
+
+/// Set the Vertical Swing mode of the A/C.
+/// @param[in] on true, the setting is on. false, the setting is off.
+/// @note This is a button press, and not a state
+///       after sending it once you should turn it off
+void IRCoronaAc::setSwingV(const bool on) {
+  setBit(&remote_state[kCoronaAcSectionData0Pos],
+         kCoronaAcSwingVToogleOffset, on);
+}
+
+/// Get the Vertical Swing mode of the A/C.
+/// @return true, the setting is on. false, the setting is off.
+bool IRCoronaAc::getSwingV(void) {
+  return GETBIT64(remote_state[kCoronaAcSectionData0Pos],
+                  kCoronaAcSwingVToogleOffset);
+}
+
+/// Set the Timer time
+/// @param[in] section index of section, used for offset.
+/// @param[in] nr_of_mins Number of minutes to set the timer to.
+///   (non in range value is disable).
+///   Valid is from 1 minute to 12 hours
+void IRCoronaAc::_setTimer(const uint8_t section, const uint16_t nr_of_mins) {
+  // default to off
+  uint16_t hsecs = kCoronaAcTimerOff;
+  if (1 <= nr_of_mins && nr_of_mins <= kCoronaAcTimerMax)
+    hsecs = nr_of_mins * kCoronaAcTimerUnitsPerMin;
+
+  uint8_t pos = section * kCoronaAcSectionBytes;
+  remote_state[pos + kCoronaAcSectionData1Pos] = (hsecs & 0xff00) >> 8;
+  remote_state[pos + kCoronaAcSectionData0Pos] = (hsecs & 0xff);
```

Comment:
ditto.
`remote_state[pos + kCoronaAcSectionData0Pos] = hsecs;` as it is being squished into a `uint8_t` anyway.